### ISSUE TITLE
Migration to Helm charts status

### ISSUE THREAD

**Issue by ohaiwalt** [prometheus-operator/kube-prometheus#29]
I saw that moving kube-prometheus over to Helm is on the roadmap in the README, is there a discussion on this anywhere or WIP? I'd be interested in helping out in whatever capacity. I'm getting close to starting this for my own use

**Comment by brancz** [comment]
We have not started anything in that direction yet and as far as I know no one else has either. Help would be greatly appreciated! 🙂

**Comment by ohaiwalt** [comment]
I was looking at that based on the current deploy scripts, and was considering using Helm hooks (pre-install, post-install) to see if that works. It should be relatively quick to test viability.

**Comment by brancz** [comment]
Sounds great. We're moving into territory where the scripts won't do anymore when we think of adding `Ingress` objects, because the external url needs to be added to the `Alertmanager` and `Prometheus` manifests if they are not served from root of a dns name. (and then we can go down the rabbit hole of securing those ingress endpoints or setting up an ingress controller in the first place, etc...) Happy to help, just let us know 🙂 .

**Comment by ohaiwalt** [comment]
Helm is currently broken on TPRs, so I'll be revisiting this after https://github.com/kubernetes/helm/issues/1468

**Comment by technosophos** [comment]
We updated #1468 -- we think everything is working except for the apply logic, which is broken in kubectl. Does this get you far enough to unblock you?

**Comment by mgoodness** [comment]
Decided to tackle this as well. Stuck on having Helm create the Operator, then waiting until the TPRs are defined before requesting Prometheus & Alertmanager instances. Hooks don't seem to do the trick. Work continues...

**Comment by mgoodness** [comment]
Currently leaving creation of the actual Alertmanager and Prometheus as manual steps until the upstream Kubernetes/Helm issues are resolved. Until then, the chart outputs a command for users to run that pipes templated TPR manifests into kubectl.

[Repo here](https://github.com/mgoodness/kube-prometheus-chart)

**Comment by brancz** [comment]
Nice job @mgoodness . Is there an idea on upstream helm to solve this problem?

**Comment by mgoodness** [comment]
Helm v2.2.0 was released yesterday, with much-improved support for TPRs. Resuming my work on the chart.

**Comment by brancz** [comment]
Awesome!

**Comment by nicgirault** [comment]
Could we have an update on this work?
